tests/unit/cmd/sca/test_precommit.py

```diff
@@ -0,0 +1,171 @@
+                                cve_ids=["CVE-2023"],
+                                identifier="GHSA-abcd-1234-xxxx",
+                            )
+                        ],
+                    )
+                ],
+            )
+        ],
+        removed_vulns=[
+            SCALocationVulnerability(
+                location="Pipfile.lock",
+                package_vulns=[
+                    SCAVulnerablePackageVersion(
+                        package_full_name="bar",
+                        version="4.5.6",
+                        ecosystem="pypi",
+                        vulns=[
+                            SCAVulnerability(
+                                severity="low",
+                                summary="another vuln",
+                                cve_ids=["CVE-2023-bis"],
+                                identifier="GHSA-efgh-5678-xxxx",
+                            )
+                        ],
+                    )
+                ],
+            )
+        ],
+    )
+
+    # Create a repo
+    repo = Repository.create(tmp_path)
+    repo.create_commit()  # Add an initial commit, if not, ref HEAD does not exist
+
+    # Writes a file
+    file_with_vulns = Path(tmp_path / "Pipfile.lock")
+    file_with_vulns.write_text("")
+
+    # Add it to git
+    repo.add(str(file_with_vulns))
+
+    with cd(str(tmp_path)):
+        result = cli_fs_runner.invoke(
+            cli,
+            [
+                "sca",
+                "scan",
+                "pre-commit",
+            ],
+        )
+
+        assert result.exit_code == ExitCode.SCAN_FOUND_PROBLEMS
+
+        # Output on added vuln
+        assert "> Pipfile.lock: 1 incident detected" in result.stdout
+        assert (
+            """
+Severity: Critical
+Summary: a vuln
+No fix is currently available.
+CVE IDs: CVE-2023"""
+            in result.stdout
+        )
+
+        # Output on removed vuln
+        assert "> Pipfile.lock: 1 incident removed" in result.stdout
+        assert (
+            """
+Severity: Low
+Summary: another vuln
+No fix is currently available.
+CVE IDs: CVE-2023-bis"""
+            in result.stdout
+        )
+
+
+@my_vcr.use_cassette("test_sca_scan_pre_commit_all.yaml")
+def test_pre_commit_all(dummy_sca_repo, cli_fs_runner):
+    # GIVEN a repo
+    dummy_sca_repo.git("checkout", "branch_with_vuln")
+    dummy_sca_repo.create_commit("No files on this one")
+    # With it's first commit with vulns
+    # And a second one without
+    # And a new file
+    (dummy_sca_repo.path / "package-lock.json").touch()
+    dummy_sca_repo.add()
+
+    # WHEN scanning with `all` flag
+    with cd(str(dummy_sca_repo.path)):
+        result = cli_fs_runner.invoke(
+            cli,
+            ["sca", "scan", "pre-commit", "--all"],
+        )
+        assert result.exit_code == ExitCode.SCAN_FOUND_PROBLEMS, result
+        assert "1 incident detected" in result.stdout
+    # THEN we get a vulnerability
```

Comment:
nit: This seems more readable to me, and is more consistent with the other comments.
```suggestion
    # THEN we get a vulnerability
    assert result.exit_code == ExitCode.SCAN_FOUND_PROBLEMS, result
    assert "1 incident detected" in result.stdout
```

Reply:
Indeed, that's a mistake